chain-api/src/types/TokenBalance.ts

```diff
@@ -425,19 +425,27 @@ export class TokenBalance extends ChainObject {
     this.lockedHolds = [...this.getUnexpiredLockedHolds(hold.created), hold];
   }
 
-  private isMatchingHold(hold: TokenHold, name?: string, lockAuthority?: string): boolean {
+  private isCallingUserAuthorized(
+    hold: TokenHold,
+    name?: string,
+    callingUser?: string,
```

Comment:
I think we can make it safer at type level by:
```suggestion
    callingUser?: UserAlias,
```

this way we can enforce on developers to make proper resolution of user alias before. User alias is unambiguous for the user, while string or user ref may be ambiguous.